### DISCUSSION TITLE
Separate light / dark mode themes

### DISCUSSION THREAD

**Discussion by RobinClowers** [chakra-ui/chakra-ui#7089] [Q&A]
The docs suggest the `useColorModeValue` hook as the solution to changing colors based on the user's color mode. This doesn't seem like a great solution to me, I would rather have separate themes for light and dark. However, it doesn't seem like there is an easy way to switch themes based on the color mode. I tried this:

```jsx
<ColorModeProvider options={lightTheme.config}>
  <ChakraProviderWrapper>
    {children}
  </ChakraProviderWrapper>
</ColorModeProvider>
```

Where ChakraProviderWrapper looks like this:

```jsx
const ChakraProviderWrapper = ({ children }: { children: React.ReactNode }) => {
  const mode = useColorMode();
  const theme = mode.colorMode === "dark" ? darkTheme : lightTheme;
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      <CSSReset />
      <EnvironmentProvider>{children}</EnvironmentProvider>
    </ThemeProvider>
  );
};
````

But the `useColorMode` hook doesn't seem to trigger a re-render when I change my color mode.

Am I doing something wrong here? Or is there some other supported way of doing this?

**Comment by yardenxr** [comment]
i had the same issue i solved it by using the "setColorMode" from "useColorMode" in order to change it reactively.

**Comment by RobinClowers** [comment]
@yardenxr thanks for reminding me of this issue. I solved this by creating my own Chakra provider, it works great:

```typescript
import {
  CSSReset,
  EnvironmentProvider,
  GlobalStyle,
  ThemeProvider,
  ToastProvider,
  useColorMode,
} from "@chakra-ui/react";
import React from "react";

import { darkTheme, lightTheme } from "~/theme";

export const ChakraProviderWrapper = ({
  children,
}: {
  children: React.ReactNode;
}) => {
  const mode = useColorMode();
  const theme = mode.colorMode === "dark" ? darkTheme : lightTheme;

  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      <CSSReset />
      <EnvironmentProvider>
        {children}
        <ToastProvider />
      </EnvironmentProvider>
    </ThemeProvider>
  );
};
```
